test/common/dashboards_data.go

```diff
@@ -0,0 +1,357 @@
+package common
+
+import (
+	goctx "context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type dashboardSearchResponse struct {
+	Title string `json:"title"`
+	UID   string `json:"uid"`
+}
+
+type targetDefinition struct {
+	Expression string `json:"expr"`
+}
+
+type panelDefinition struct {
+	Datasource string             `json:"datasource"`
+	Targets    []targetDefinition `json:"targets"`
+}
+
+type dashboardDefinition struct {
+	Panels []panelDefinition `json:"panels"`
+}
+
+type dashboardDetailResponse struct {
+	Dashboard dashboardDefinition `json:"dashboard"`
+}
+
+type prometheusQueryResult struct {
+	Metric map[string]interface{} `json:"metric"`
+	Value  []interface{}          `json:"value"`
+}
+
+type prometheusQueryData struct {
+	Result []prometheusQueryResult `json:"result"`
+}
+
+type prometheusQueryResponse struct {
+	Status string              `json:"status"`
+	Data   prometheusQueryData `json:"data"`
+}
+
+var (
+	expectedServices = [...]string{
+		"3scale-admin-ui",
+		"3scale-developer-console-ui",
+		"apicurito-ui",
+		"codeready-ui",
+		"amq-service-broker",
+		"rhsso-ui",
+		"rhssouser-ui",
+		"3scale-system-admin-ui",
+		"webapp-ui",
+		"syndesis-ui",
+		"ups-ui",
+	}
+	dashboardsNames = [...]string{
+		"Endpoints Summary",
+		"Endpoints Detailed",
+		"Endpoints Report",
+		"Resource Usage By Pod",
+		"Resource Usage for Cluster",
+		"Resource Usage By Namespace",
+	}
+)
+
+// TestDashboardsData verifies that all dashboards are installed and all the graphs are filled with data
+func TestDashboardsData(t *testing.T, ctx *TestingContext) {
+	grafanaPodName, err := getMonitoringAppPodName("grafana", ctx)
+	if err != nil {
+		t.Fatal("failed to get grafana pod name", err)
+	}
+
+	prometheusPodName, err := getMonitoringAppPodName("prometheus", ctx)
+	if err != nil {
+		t.Fatal("failed to get prometheus pod name", err)
+	}
+
+	// retry the tests every minute for up to 10 minutes
+	monitoringTimeout := 10 * time.Minute
```

Comment:
@jhellar what are potential reasons for a failure that means a retry would succeed later?

Reply:
When running the test in prow, I found out that not all queries return values shortly after the RHMI installation finished. So this is just to make sure that there is enough time for all the queries to return some values.